File: src/main/java/com/kamis/Car_Rental_Spring/services/admin/AdminService.java
package com.kamis.Car_Rental_Spring.services.admin;

import com.kamis.Car_Rental_Spring.dto.CarDto;

import java.io.IOException;
import java.util.List;

public interface AdminService {

    boolean postCar(CarDto carDto) throws IOException;

    List<CarDto> getAllCars();

    void deleteCar(Long id);
}

File: src/app/modules/admin/services/admin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Observable} from "rxjs";
import {StorageService} from "../../../auth/services/storage/storage.service";

const BASIC_URL = ["http://localhost:8080"];

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor( private http: HttpClient) { }

  createAuthorizationHeader(): HttpHeaders {
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set('Authorization', 'Bearer '+StorageService.getToken());
  }

  postCar(carDto:any):Observable<any>{
    return this.http.post(BASIC_URL+"/api/admin/car", carDto, {
      headers: this.createAuthorizationHeader()});
  }

  getAllCars():Observable<any> {
    return this.http.get(BASIC_URL+"/api/admin/cars", {
      headers: this.createAuthorizationHeader()
    });

  }


}
